python/ray/serve/_private/application_state.py

```diff
@@ -454,38 +454,13 @@ def _determine_app_status(self) -> Tuple[ApplicationStatus, str]:
         if self._target_state.deleting:
             return ApplicationStatus.DELETING, ""
 
-        num_healthy_deployments = 0
-        num_autoscaling_deployments = 0
-        num_updating_deployments = 0
-        num_manually_scaling_deployments = 0
-        unhealthy_deployment_names = []
-
-        for deployment_status in self.get_deployments_statuses():
-            if deployment_status.status == DeploymentStatus.UNHEALTHY:
-                unhealthy_deployment_names.append(deployment_status.name)
-            elif deployment_status.status == DeploymentStatus.HEALTHY:
-                num_healthy_deployments += 1
-            elif (
-                deployment_status.status_trigger == DeploymentStatusTrigger.AUTOSCALING
-            ):
-                num_autoscaling_deployments += 1
-            elif deployment_status.status == DeploymentStatus.UPDATING:
-                num_updating_deployments += 1
-            elif (
-                deployment_status.status
-                in [DeploymentStatus.UPSCALING, DeploymentStatus.DOWNSCALING]
-                and deployment_status.status_trigger
-                == DeploymentStatusTrigger.CONFIG_UPDATE_STARTED
-            ):
-                num_manually_scaling_deployments += 1
-            else:
-                raise RuntimeError(
-                    "Found deployment with unexpected status "
-                    f"{deployment_status.status} and status trigger "
-                    f"{deployment_status.status_trigger}."
-                )
-
-        if len(unhealthy_deployment_names):
+        choice_status = min(self.get_deployments_statuses(), key=lambda info: info.rank)
```

Comment:
nit: more self-describing
```suggestion
        lowest_rank_status = min(self.get_deployments_statuses(), key=lambda info: info.rank)
```

Reply:
A comment helping to describe the purpose of this line would also be helpful.

It may also be preferable to structure this as a static method on the `DeploymentStatus` class so the rank definitions & logic are centralized there.